scripts/zkai_paper_claim_pack_gate.py

```
@@ -234,6 +242,41 @@ def _assert_evidence_paths_exist(payload: dict[str, Any]) -> None:
             raise ClaimPackGateError(f"missing evidence path: {path}")
         _assert_repo_contained(full_path, f"evidence_refs[{index}].path")
+        if ref.get("id") == ROUTE_MATRIX_REF_ID:
+            _assert_route_matrix_integrity(full_path)
+
+
+def _assert_route_matrix_integrity(full_path: pathlib.Path) -> None:
+    with full_path.open(encoding="utf-8") as handle:
+        route_data = json.load(handle)
+    rows = route_data.get("route_rows")
+    if not isinstance(rows, list):
+        raise ClaimPackGateError("route_matrix route_rows must be a list")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)